### DISCUSSION TITLE
Instagram-like modal small issue

### DISCUSSION THREAD

**Discussion by Aminejvm** [vercel/next.js#13145] [Help]
![Capture](https://user-images.githubusercontent.com/39884652/82466698-44f43c80-9ab0-11ea-9dc8-a8ae0ac18aa8.PNG)
This is my Link component that allows such behavior, it works fine unless the route is dynamic.
----
I tried changing the CURRENT_PATH to router.asPath, but as a result, it hard refreshes to /login 
I know href property should follows the page routes structure so router.asPath is no help. 
Is this impossible to do?
```javascript
const currentPath = "profile/[id]" 
<Link href={`${currentPath}?query=`something`} as="/login" />  
```

----- 
I'm using the auth query to toggle between signup and signing modal. 
Also to close the modal I use router.push(router.pathname)

**Comment by ghost** [comment]
This video should help you.

https://www.youtube.com/watch?v=EyjuRCxpf6Y
